pkg/webhook/policy.go

```diff
@@ -182,23 +185,35 @@ func (h *validationHandler) Handle(ctx context.Context, req admission.Request) a
 	}
 
 	res := resp.Results()
-	msgs := h.getDenyMessages(res, req)
-	if len(msgs) > 0 {
-		vResp := admission.ValidationResponse(false, strings.Join(msgs, "\n"))
+	denyMsgs, warnMsgs := h.getValidationMessages(res, req)
+
+	if len(denyMsgs) > 0 {
+		vResp := admission.ValidationResponse(false, strings.Join(denyMsgs, "\n"))
 		if vResp.Result == nil {
 			vResp.Result = &metav1.Status{}
 		}
+		if len(warnMsgs) > 0 {
+			vResp.Warnings = warnMsgs
+		}
 		vResp.Result.Code = http.StatusForbidden
 		requestResponse = denyResponse
 		return vResp
 	}
 
 	requestResponse = allowResponse
-	return admission.ValidationResponse(true, "")
+	vResp := admission.ValidationResponse(true, "")
+	if vResp.Result == nil {
+		vResp.Result = &metav1.Status{}
+	}
+	if len(warnMsgs) > 0 {
+		vResp.Warnings = warnMsgs
+		vResp.Result.Code = httpStatusWarning
```

Comment:
I think if any denies are present, `http.StatusForbidden` should take precedence. I don't think K8s is relying on the status code, but it's good to follow conventions.

Reply:
wait, NVM, I misread the code.